src/mem/queue.rs

```diff
@@ -69,6 +69,13 @@ impl RollingBuffer {
             Cow::Owned(res)
         }
     }
+
+    fn shrink_if_sparse(&mut self) {
+        let capacity = self.buffer.capacity();
+        if self.buffer.len() < capacity / 2 && capacity > 10 {
+            self.buffer.shrink_to(capacity / 2);
+        }
```

Comment:
what about:
```suggestion
        if self.buffer.len() * 2 < capacity {
            self.buffer.shrink_to_fit();
        }
```

Reply:
`capacity / 2` isn't ideal, but I believe using shrink_to_fit will likely cause the buffer size to double at next truncation: there is no headroom, if we receive one byte more this time, we have to reallocate. Maybe something like len *1.125 would be better? It gives enough headroom to likely not reallocate next round, while still shrinking the queue substantially

Reply:
I don't think `1.125` would help much.

The way I understand it, 
- we append data
- we truncate to a position that is most of the time not the tail.
the remaining `len` is not a great estimator of what the right capacity should look like.

One thing interesting however is, if everything works as intended, we are guaranteed to get another call to truncate in the future if the len is > 0.

We should probably shrink to fit for len = 0.
That's the common case of the queue that is not deleted but is not getting new documents either.

For len > 0, we can do `capacity / 2` I think. If it is too wasteful (if len is suddenly very), it will still be halved in the next suggest anyway.